e2e-tests/playwright/e2e/plugins/custom-theme/theme-verifier.ts

```diff
@@ -0,0 +1,83 @@
+import { Page, expect, TestInfo } from '@playwright/test';
+import { UIhelper } from '../../../utils/UIhelper';
+import { UIhelperPO } from '../../../support/pageObjects/global-obj';
+
+export class ThemeVerifier {
+  private readonly page: Page;
+  private uiHelper: UIhelper;
+  private readonly tasks: Array<() => Promise<void>>;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.uiHelper = new UIhelper(page);
+    this.tasks = [];
+  }
+
+  setTheme(theme: 'Light' | 'Dark'): this {
+    this.tasks.push(async () => {
```

Comment:
Are there any advantages/reasons to push functions in an array and executing them later instead of just running them right away?

Reply:
@subhashkhileri  The advantage is that we don't need to add `await` em every single line in our tests, look the way that the test are invoking the code:
```
await themeVerifier
      .setTheme('Dark')
      .verifyHeaderGradient(
        'none, linear-gradient(90deg, rgb(0, 0, 208), rgb(255, 246, 140))',
      )
      .verifyBorderLeftColor('rgb(244, 238, 169)')
      .takeScreenshotAndAttach(
        'screenshots/custom-theme-dark-inspection.png',
        testInfo,
        'custom-theme-dark-inspection',
      )
```
IMHO Fluent interfaces create code that reads more like natural language. This enhances the readability and understanding of the tests, especially for non-developers or stakeholders who need to comprehend what the tests are doing.  
The fluent interface pattern helps in grouping related operations together, making the code easier to maintain. When operations are logically grouped, understanding and modifying the code becomes simpler, reducing the risk of introducing errors.
By enabling method chaining, fluent interfaces help in reducing boilerplate code. This means fewer lines of code to achieve the same functionality, leading to more efficient and streamlined test scripts.
Modern IDEs provide better support for method chaining with intellisense and autocompletion features. This can speed up development by reducing the time spent looking up method names and parameters.

Reply:
Fluent design is subjective. In my POV it increases the complexity as one has to push the method along with test actions to array, has to return the class reference on every method and then has to make sure one called `.execute()` to make sure those methods got executed.

Reply:
and I am not sure, if it makes hard to debug, as the anonymous methods are called from array in a loop.

Reply:
> and I am not sure, if it makes hard to debug, as the anonymous methods are called from array in a loop.

@subhashkhileri the Error messages comes normal from the ThemeVerfier, you can now the exactly lines which invokes the error